### ISSUE TITLE
Rotary Encoder

### ISSUE THREAD

**Issue by MConnorR** [The-Mad-Noodle/Mad-Noodle-Via-Support#3]
I'm having a issue attaching a Macro to the rotary encoder options 
its to use short cuts with drawing programs on a ipad if that helps

I'm trying to copy and paste the mapped macro
{+KC_LGUI}{+KC_LBRC}{-KC_LGUI}
and it won't alow it to be comfirmed. i watched the set up video and the way he explained to type it also hasn't worked for me.

I am using Via on a Mac so the windows button is the command button not sure if that matters.

Appreciate any help,

MC

**Comment by jessel92** [comment]
Hey there! 

I'm super sorry to hear that you're having trouble with your keypad! 

The first issue I noticed is that you're not "releasing" the LBRAC keycode. The + sign indicates pushing down the key, while the - sign indicates releasing it. With your current macro, you're constantly holding down the bracket key. Try removing the + before the bracket. If there's no sign in front of the keycode, it means "tap", which is a quick key press. So it would translate to Hold down CMD, tap bracket, Release the CMD key. 

Another option to try is adding a {-KC_LBRC} at the end to release the bracket key, like this: {+KC_LGUI}{+KC_LBRC}{-KC_LGUI}{-KC_LBRC}. Then do the same for the opposite rotation direction, but use RBRC instead. 

In simpler terms, the encoder activates a keycode every time you feel the "click" as you rotate it. These encoders are 24PPR, which means they activate a keycode 24 times for every full 360-degree revolution. Your goal is to activate CMD+Bracket to change your brush size, so you need to do just that. You just need to make sure it completes the key combo before its next click. 

But an even simpler way of doing ALL of this, rather than using macros, is to go directly to the encoder and enter LGUI(KC_LBRC) for CCW and LGUI(KC_RBRC) for CW. It's just a simplified way of pressing the CMD+Bracket combo! 

Hope this helps! 
Let me know if that works for you!
